dev/AppLifecycle/ExtensionContract.h

```diff
@@ -2,31 +2,52 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 #pragma once
 
+#include "winrt/Microsoft.Windows.AppLifecycle.h"
 #include "LaunchActivatedEventArgs.h"
 #include "ProtocolActivatedEventArgs.h"
 #include "FileActivatedEventArgs.h"
 #include "StartupActivatedEventArgs.h"
 
-namespace winrt::Microsoft::ApplicationModel::Activation::implementation
+namespace winrt::Microsoft::Windows::AppLifecycle::implementation
 {
-    // Extension contract ids.
-    static LPCWSTR c_launchContractId = L"Windows.Launch";
-    static LPCWSTR c_fileContractId = L"Windows.File";
-    static LPCWSTR c_protocolContractId = L"Windows.Protocol";
-    static LPCWSTR c_startupTaskContractId = L"Windows.StartupTask";
-
     // This array holds the mapping between a class factory and it's extension contract Id.
     static const struct ExtensionMap
     {
+        ExtendedActivationKind kind;
         std::wstring contractId;
-        Windows::ApplicationModel::Activation::IActivatedEventArgs(*factory)(
-            Windows::ApplicationModel::Activation::IProtocolActivatedEventArgs const& args);
+        winrt::Windows::ApplicationModel::Activation::IActivatedEventArgs(*factory)(winrt::Windows::Foundation::Uri const& uri);
     }
     c_extensionMap[] =
     {
-        { c_launchContractId, &LaunchActivatedEventArgs::CreateFromProtocol },
-        { c_fileContractId, &FileActivatedEventArgs::CreateFromProtocol },
-        { c_protocolContractId, &ProtocolActivatedEventArgs::CreateFromProtocol },
-        { c_startupTaskContractId, &StartupActivatedEventArgs::CreateFromProtocol },
+        { ExtendedActivationKind::Launch, c_launchContractId, &LaunchActivatedEventArgs::Deserialize },
+        { ExtendedActivationKind::File, c_fileContractId, &FileActivatedEventArgs::Deserialize },
+        { ExtendedActivationKind::Protocol, c_protocolContractId, &ProtocolActivatedEventArgs::Deserialize },
+        { ExtendedActivationKind::StartupTask, c_startupTaskContractId, &StartupActivatedEventArgs::Deserialize },
     };
+
+    inline bool IsEncodedLaunch(winrt::Windows::Foundation::Uri const& uri)
+    {
+        return CompareStringOrdinal(uri.SchemeName().c_str(), -1, c_encodedLaunchSchemeName, -1, TRUE) == CSTR_EQUAL;
+    }
+
+    inline std::tuple<ExtendedActivationKind, winrt::Windows::Foundation::IInspectable> DecodeActivatedEventArgs(winrt::Windows::Foundation::Uri const& uri)
+    {
+        for (auto const& pair : uri.QueryParsed())
+        {
+            if (CompareStringOrdinal(pair.Name().c_str(), -1, c_contractIdKeyName, -1, TRUE) == CSTR_EQUAL)
+            {
+                for (int index = 0; index < _countof(c_extensionMap); index++)
```

Comment:
If you want to get fancy, you can use range-based for:

```c++
for (const auto& extension : c_extensionMap) {
    /* use "extension" instead of "c_extensionMap[index]" in the body */
}
```